tests/unit/Commands/ContinuousIntegration/TravisCi/Config/ConfigWriterTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace QuickStrapUnit\Commands\ContinuousIntegration\TravisCi\Config;
+
+use org\bovigo\vfs\vfsStream;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\Config;
+use QuickStrap\Commands\ContinuousIntegration\TravisCi\Config\ConfigWriter;
+use Symfony\Component\Yaml\Yaml;
+
+class ConfigWriterTest extends \PHPUnit_Framework_TestCase
+{
+    /** @var  string */
+    private $configFile;
+    
+    /** @var ConfigWriter */
+    private $sut;
+
+    protected function setUp()
+    {
+        parent::setUp();
+
+        vfsStream::setup('application_path')
+            ->addChild($configFile = vfsStream::newFile('.travis.yml'));
+        
+        $this->configFile = $configFile->url();
+
+        $this->sut = new ConfigWriter();
+    }
+    
+    public function test_can_write_language_to_yml()
+    {
+        $config = new Config('php');
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals("language: php\n", $actual);
+    }
+    
+    public function test_can_write_php_versions_to_yml()
+    {
+        $config = new Config('php');
+        $config->setPhp(['7.0', '5.6']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+php:
+    - '7.0'
+    - '5.6'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_before_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setBeforeScript(['composer self-update', 'composer install']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+before_script:
+    - 'composer self-update'
+    - 'composer install'
+
+YML
+, $actual);
+    }
+    
+    public function test_can_write_script_to_yml()
+    {
+        $config = new Config('php');
+        $config->setScript(['vendor/bin/phpunit', 'vendor/bin/behat']);
+
+        $actual = $this->sut->toYml($config);
+
+        self::assertEquals(<<<YML
+language: php
+script:
+    - vendor/bin/phpunit
+    - vendor/bin/behat
+
+YML
+, $actual);
+    }
+    
+    public function test_can_allow_failures_to_yml()
```

Comment:
Imagine you had a class that generated a comma separated value line from an array of values. Ie, `Csv::line(array $values)`. Lets say you were creating a report class. You have a requirement that report class should write out lines with `product,price,qty`. 

You have two options to test your report class. You can do a unit test, where you mock `Csv::line` and expect it to be called with an array `['bat','$3.00',4]`. There is an implicit trust that the format provided by the class to the writer (the array) is correct. There is also an implicit linking that the report class is generating a format that is specifically designed for Csv::line to consume. 

Or you can write an integration test where you assert the contents of a file/stream contain `bat,$3.00,4`. 

This is the same scenario of your `Config` and `ConfigWriter`. You actually do know the specs of the YAML format (you can do Yaml::parse and print_r to see the array structure that is represented. You can make assertions against Config that test when you add a failure that the expected array structure is created. Again, implicitly you are linking config to Yaml generation (I'm not necessarily saying that is a bad thing). 

Otherwise, we can accept the brittleness of the integration test.